### PR TITLE
add javadocs to the UncheckedPredicate class

### PR DESCRIPTION
**Pull request by acoburn**
@ajs6f may have some comments on the javadocs here....

@yinlinchen whether you choose to merge this PR or not, you should rebase this branch w/r/t `fcrepo4/master`. One reason your last build failed is due to the fact that this uses an older version of the `checkstyle` plugin. But since that change has already been merged into master, a simple rebase will fix it:

    git fetch upstream
    git rebase upstream/master

### CODE REVIEW

**Comment by yinlinchen** [comment]
I will do it sometime late tomorrow. Thank you!

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java [hunk 1]
```diff
@@ -22,9 +22,20 @@
...
+ */
 @FunctionalInterface
 public interface UncheckedPredicate<T> extends Predicate<T>, com.google.common.base.Predicate<T> {
 
+    /**
+     *  Test this predicate against the given object.
+     *
+     *  @param elem an item of type T
+     *  @return the value of this predicate when applied to the input elem.
+     */
```
Comment (ajs6f) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java]: You should not have to supply Javadocs for `@Override`s, and in this case I don't think it clarifies the intent. I'd leave these out.

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java [hunk 2]
```diff
@@ -34,6 +45,12 @@ default boolean test(final T elem) {
         }
     }
 
+    /**
+     *  Apply this predicate to the given object.
+     *
+     *  @param elem an item of type T
+     *  @return the value of this predicate when applied to the input elem.
+     */
```
Comment (ajs6f) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java]: You should not have to supply Javadocs for `@Override`s, and in this case I don't think it clarifies the intent. I'd leave these out.

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java [hunk 3]
```diff
@@ -43,9 +60,22 @@ default boolean apply(final T elem) {
...
     boolean testThrows(T elem) throws RepositoryException;
 
+    /**
+     *  Determine a true/false value for each item in a collection of inputs,
+     *  without warning about type checks.
+     *
+     *  @param p an unchecked predicate
+     *  @return the predicate given to this function
+     */
```
Comment (ajs6f) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java]: I do not understand the doc here at all. `Predicate`s have no inherent connection with collections. How about simply "A static convenience method for constructing UncheckedPredicates from lambda syntax."?

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java [hunk 4]
```diff
@@ -43,9 +60,22 @@ default boolean apply(final T elem) {
         }
     }
 
+    /**
+     *  Test a throwable condition
+     *
+     *  @param elem an item of type T
+     *  @return the value of this predicate when applied to the input element.
+     */
     boolean testThrows(T elem) throws RepositoryException;
```
Comment (ajs6f) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java]: It's not clear here what a "throwable condition" is. How about "This method has the same semantic as {@link #test} but permits {@link RepositoryException}s."?

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java (hunk 3 above)
Comment (acoburn) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java]: Thanks, that's a much clearer way of expressing that

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java [hunk 5]
```diff
@@ -22,9 +22,20 @@
 
 import org.fcrepo.kernel.exception.RepositoryRuntimeException;
 
+/**
+ * A functional, unchecked predicate for any collection<T>
+ *
+ * @author ajs6f
+ */
```
Comment (ajs6f) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/UncheckedPredicate.java]: Collections are not relevant here. How about just "A {@link Predicate} that permits {@link RepositoryException}s."?

**Comment by yinlinchen** [comment]
@acoburn I did rebase, merged code, issued mvn clean install in my local and all test passed. However, I just check the CI build in fcrepo4, it is still failed.....

**Comment by acoburn** [comment]
@yinlinchen travis is failing on javadoc errors. Try running

    mvn javadoc:jar

And fix any errors in the `UncheckedPredicate` class. I think it's mostly a matter of changing `{@RepositoryException}s.<T>` to `{@link RepositoryException}s.`

**Comment by acoburn** [comment]
@yinlinchen actually, what you want to run is:

    mvn javadoc:test-aggregate

**Pull request merged**